fetcher/dockerhub.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+#
+# ** License **
+#
+# Home: http://resin.io
+#
+# Author: Andrei Gherzan <andrei@resin.io>
+#
+
+import os
+import logging
+from docker import Client
+from fetcher.tar import tarFetcher
+from modules.util import *
+
+log = logging.getLogger(__name__)
+
+class dockerhubFetcher(tarFetcher):
+
+    def __init__ (self, conffile, version, remote):
+        super().__init__()
+        self.remotefile = os.path.join(self.remote + ":resinhup-" + machine + "-" + version)
+        self.workspacefile = os.path.join(self.workspace, "resinhup.tar")
+
+    def download(self):
+        self.cleanworkspace()
+
+        container_name='resinhup'
+
+        try:
+            cli = Client(base_url='unix://var/run/docker.sock', version='auto')
+        except:
+            log.error("Can't connect to docker daemon.")
+            return False
+
+        # Pull docker image
+        try:
+            log.info("Docker image pull started... this can take a couple of minutes...")
+            log.debug("Pulling " + self.remotefile + " ...")
+            cli.pull(self.remotefile, stream=False)
+        except:
+            log.error("Can't pull update image.")
+            return False
+
+        # Make sure there is no 'resinhup' container
+        try:
+            cli.remove_container(container_name, force=True)
+        except:
+            pass
+
+        # Create resinhup container
+        try:
+            # Docker mounts a set of files at container creation
+            # https://docs.docker.com/engine/userguide/networking/default_network/configure-dns/
+            # We still need all these files as they are important in the update process
+            cmd='bash -c "umount /etc/resolv.conf ; umount /etc/hostname ; umount /etc/hosts"'
```

Comment:
This was an old implementation for the fix of hostname and co shadowing. I forgot to remove it. Good catch.